lgc/patch/NggPrimShader.cpp

```diff
@@ -79,7 +79,7 @@ enum {
 NggPrimShader::NggPrimShader(PipelineState *pipelineState)
     : m_pipelineState(pipelineState), m_context(&pipelineState->getContext()),
       m_gfxIp(pipelineState->getTargetInfo().getGfxIpVersion()), m_nggControl(m_pipelineState->getNggControl()),
-      m_ldsManager(nullptr), m_constPositionZ(false), m_builder(new IRBuilder<>(*m_context)) {
+      m_builder(std::make_unique<IRBuilder<>>(pipelineState->getContext())) {
```

Comment:
While you're at it, why is `m_builder` a unique pointer in the first place? Couldn't it be a plain `IRBuilder<>` object and avoid the heap allocation?

On second thought, this adds a whole lot of additional churn, but if you could do it anyway (maybe in a separate change) I'd appreciate it.

Reply:
Thank you very much. I will try to correct it in a separate change. I plan to do a series of changes to refine NGG coding in these weeks to get rid of unclear coding issues and vague design logic.

Reply:
I made an extra change to correct the use of IR builder.